Tests/KituraNetTests/KituraNetWebSocketUpgrade.swift

```diff
@@ -0,0 +1,175 @@
+            expectation.fulfill()
+            
+        },{expectation in
+            let upgraded = DispatchSemaphore(value: 0)
+            self.sendUpgradeRequest(toPath: "/", usingKey: "test", wsVersion: "13", semaphore: upgraded)
+            upgraded.wait()
+            XCTAssertEqual(self.httpHandler!.responseStatus, .badRequest, "Test case failed as status code    \(self.httpHandler!.responseStatus) was returned instead of badRequest")
+            expectation.fulfill()
+            
+        },{expectation in
+            let upgraded = DispatchSemaphore(value: 0)
+            self.sendUpgradeRequest(toPath: "/", usingKey: "test", wsVersion: "1", semaphore: upgraded)
+            upgraded.wait()
+            XCTAssertEqual(self.httpHandler!.responseStatus, .badRequest, "Test case failed as status code \(self.httpHandler!.responseStatus) was returned instead of badRequest")
+            expectation.fulfill()
+            
+        })
+}
+    
+    class WebSocketUpgradeDelegate: ServerDelegate {
+        func handle(request: ServerRequest, response: ServerResponse) {}
+    }
+    
+    func sendUpgradeRequest(toPath: String, usingKey: String, wsVersion: String, semaphore: DispatchSemaphore, errorMessage: String? = nil) {
+        
+        self.httpHandler = HTTPResponseHandler(key: usingKey,semaphore: semaphore, errorMessage: errorMessage)
+        let clientBootstrap = ClientBootstrap(group: MultiThreadedEventLoopGroup(numberOfThreads: 1))
+            .channelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEPORT), value: 1)
+            .channelInitializer(clientChannelInitializer)
+        
+        do {
+            let channel = try clientBootstrap.connect(host: "localhost", port: self.port).wait()
+            var request = HTTPRequestHead(version: HTTPVersion(major: 1, minor: 1), method: HTTPMethod(rawValue: "GET"), uri: toPath)
+            var headers = HTTPHeaders()
+            headers.add(name: "Upgrade", value: "websocket")
+            headers.add(name: "Connection", value: "Upgrade")
+            headers.add(name: "Sec-WebSocket-Version", value: wsVersion)
+            headers.add(name: "Sec-WebSocket-Key", value: usingKey)
+            request.headers = headers
+            channel.write(NIOAny(HTTPClientRequestPart.head(request)), promise: nil)
+            try channel.writeAndFlush(NIOAny(HTTPClientRequestPart.end(nil))).wait()
+        } catch let error {
+            Log.error("Error: \(error)")
+    }
+}
+}
+
+class HTTPResponseHandler: ChannelInboundHandler {
+    
+    public typealias InboundIn = HTTPClientResponsePart
+    public var responseStatus : HTTPResponseStatus
+
+    let errorMessage: String?
+    let key: String
+    let upgradeDoneOrRefused: DispatchSemaphore
+    
+    public init(key: String, semaphore: DispatchSemaphore, errorMessage: String? = nil) {
```

Comment:
We aren't using `errorMessage` anywhere in `HTTPResponseHandler`. May be there's no need to store it here?